### PR TITLE
Visual Design Cleanup for Headers - Filters and header buttons

### PR DESCRIPTION
**Pull request by akiselev**
See the [notion doc](https://www.notion.so/metabase/Visual-Design-Cleanup-for-Headers-3475a1338cb840f28904063553b8ed87) and #19909 

**Repro Steps**
* Create a question
* Sample Dataset > Product
* Add a filter Price less than 30

**Before**
image

image

**After**
image

image

### CODE REVIEW

**Review by gusaiani — APPROVED** [review]
Code looks pretty good and so does the UI.

One tweak I'd suggest: optically centering the text in the Filter and Summarize buttons.

padding-top: 12px;
padding-bottom: 14px;

seem to do it.

image

**Comment by kulyk** [comment]
Looks cool! Noticed that "Filter" and "Summarize" buttons loose the hover outline after you open and then close them, see:

CleanShot 2022-01-27 at 19 11 46

File: frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx [hunk 1]
```diff
@@ -53,3 +54,38 @@ export const DatasetHeaderButtonContainer = styled.div`
...
+export const HeaderButton = styled(Button)`
+  background-color: ${({ active, color = getDefaultColor() }) =>
+    active ? alpha(color, 0.8) : "transparent"};
+  color: ${({ active }) => (active ? "white" : color("text-dark"))};
+  &:hover {
+    background-color: ${({ active, color = getDefaultColor() }) =>
+      active ? alpha(color, 0.8) : "transparent"};
+    color: ${({ active }) => (active ? "white" : color("text-dark"))};
+  }
+  transition: background 300ms linear, border 300ms linear;
```
Comment (ariya) [inline frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx:67]: While this is likely subtle enough, doesn't hurt to support reduced motion (see #18169).

Comment (akiselev) [reply frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx:67]: Nice catch! Fixed

**Comment by akiselev** [comment]
> Looks cool! Noticed that "Filter" and "Summarize" buttons loose the hover outline after you open and then close them, see:
> 
> CleanShot 2022-01-27 at 19 11 46

Fixed

> Code looks pretty good and so does the UI.
> 
> One tweak I'd suggest: optically centering the text in the Filter and Summarize buttons.
> 
> padding-top: 12px; padding-bottom: 14px;
> 
> seem to do it.
> 
> image

Added to the epic as a future update

**Pull request merged**